cli/internal/cmd/upgradecheck.go

```diff
@@ -519,6 +576,61 @@ func getCompatibleImageMeasurements(ctx context.Context, writer io.Writer, clien
 	return upgrades, nil
 }
 
+// newCLIVersions returns a list of versions of the CLI which are compatible with the current Kubernetes version.
+func (v *versionCollector) newCLIVersions(ctx context.Context, currentKubernetesVersion string) ([]string, error) {
+	versionFetcher := fetcher.NewFetcher()
+
+	list := versionsapi.List{
+		Ref:         "feat-upload-cli-version-list",
+		Stream:      "debug",
+		Granularity: versionsapi.GranularityMajor,
+		Base:        "v2",
+		Kind:        versionsapi.VersionKindCLI,
+	}
+	minorList, err := versionFetcher.FetchVersionList(ctx, list)
+	if err != nil {
+		return nil, fmt.Errorf("listing minor versions: %w", err)
+	}
+
+	var patchVersions []string
+	for _, version := range minorList.Versions {
```

Comment:
No need to check all minor versions, as only the current and upcoming minor versions are relevant.